### PR TITLE
feat: create skeleton for customer ticket details

### PR DESCRIPTION
**Pull request by Parthlakhani**
JIRA ticket: https://jira.tools.sap/browse/CXEC-14757

### CODE REVIEW

File: feature-libs/customer-ticketing/occ/config/default-occ-customer-ticketing-config.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { OccConfig } from '@spartacus/core';
+
+export const defaultOccCustomerTicketingConfig: OccConfig = {
+  backend: {
+    occ: {
+      endpoints: {
+        getTicketDetails: 'users/${customerId}/tickets/${ticketId}',
```
Comment (khristian23) [inline feature-libs/customer-ticketing/occ/config/default-occ-customer-ticketing-config.ts]: OCC endpoint is named "getTicket".  Please update to synchronize with backend

https://github.tools.sap/cx-commerce/commerce-apis/blob/develop/commerce-apis-internal/Artifacts/customerticketing-internal/customerticketing-internal.yaml#L6108

File: feature-libs/customer-ticketing/occ/model/occ-customer-ticketing-endpoints.model.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { OccEndpoint } from '@spartacus/core';
+
+declare module '@spartacus/core' {
+  interface OccEndpoints {
+    /**
+     * Endpoint to get ticket details by ticket id
+     *
+     * * @member {string}
+     */
+    getTicketDetails?: string | OccEndpoint;
```
Comment (khristian23) [inline feature-libs/customer-ticketing/occ/model/occ-customer-ticketing-endpoints.model.ts]:
```suggestion
    getTicket?: string | OccEndpoint;
```

**Review by khristian23 — COMMENTED** [review]
Minor changes to match what is written in OCC

File: feature-libs/customer-ticketing/occ/config/default-occ-customer-ticketing-config.ts (hunk 1 above)
Comment (Parthlakhani) [reply feature-libs/customer-ticketing/occ/config/default-occ-customer-ticketing-config.ts]: got it

File: feature-libs/customer-ticketing/components/customer-ticketing-components.module.ts [hunk 3]
```diff
@@ -14,6 +15,7 @@ import {
     I18nModule,
     FormErrorsModule,
     PasswordVisibilityToggleModule,
```
Comment (bgambocjaviniar) [inline feature-libs/customer-ticketing/components/customer-ticketing-components.module.ts]: Do you require the passwordvisibilitytoggle module?

is there an input box that needs to be 'private'?

**Review by bgambocjaviniar — COMMENTED** [review]
LGTM,

e2e is not a problem.

Rule of thumb since there's no cron job that resets the server and it hasn't been re-initialized since it was up

If you didn't touch core spartacus features other than customer-ticketing, you can disregard the e2es.

For example, this skeleton PR does not even touch anything related to the broken e2es and is only within the customer ticketing 'domain'

File: feature-libs/customer-ticketing/components/customer-ticketing-components.module.ts (hunk 3 above)
Comment (Parthlakhani) [reply feature-libs/customer-ticketing/components/customer-ticketing-components.module.ts]: nope, will remove it.

**Review by khristian23 — APPROVED** [review]

**Pull request merged**